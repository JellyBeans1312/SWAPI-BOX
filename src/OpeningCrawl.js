import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css';

const OpeningCrawl = ({title, date, episode, text}) => {
  console.log(title)
  return (
    <Crawl 
    title = {title}
    subtitle ={episode}
    text = {text}
    />
  )
}

export default OpeningCrawl;
