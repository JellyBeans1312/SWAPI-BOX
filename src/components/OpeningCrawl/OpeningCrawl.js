import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css';
import './OpeningCrawl.scss'
import PropTypes from 'prop-types'

const OpeningCrawl = ({title, date, episode, text, hideLanding}) => {
  const romanNum = () => {
    let roman = [0, 'I', 'II', 'III', 'IV', 'V', 'VI', 'VII', 'VIII']
    return roman[episode]
  }
  return (
    <article>
      <Crawl 
      title = {`Episode ${romanNum()}`}
      subTitle={title}
      text = {text + date}
      />
      <button className="hide-movie" onClick={()=> hideLanding()}>Skip Movie Stuff</button>
    </article>
  )
}

export default OpeningCrawl;

OpeningCrawl.propTypes = {
  title: PropTypes.string.isRequired,
  episode: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  hideLanding: PropTypes.func.isRequired
}