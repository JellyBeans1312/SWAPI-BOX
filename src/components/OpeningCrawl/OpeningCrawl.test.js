import React from 'react';
import { shallow } from 'enzyme';
import OpeningCrawl from './OpeningCrawl';

describe('OpeningCrawl', () => {
  let wrapper;
  let mockHideLanding = jest.fn()
  beforeEach(() => {
    wrapper = shallow(
      <OpeningCrawl
        title='hello'
        date='something'
        episode='four'
        text='long time ago'
        hideLanding={mockHideLanding}
      />
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should call hideLanding onclick', () => {
    wrapper.find('.hide-movie').simulate('click')
    expect(mockHideLanding).toHaveBeenCalled()
  });
}); 