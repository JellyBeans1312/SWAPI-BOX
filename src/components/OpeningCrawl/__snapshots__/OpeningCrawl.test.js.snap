// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`OpeningCrawl should match snapshot 1`] = `
<article>
  <Crawl
    subTitle="hello"
    text="long time agosomething"
=======
exports[`OpeningCrawl should match a snapshot 1`] = `
<article>
  <Crawl
    text={NaN}
>>>>>>> more-testing
    title="Episode undefined"
  />
  <button
    className="hide-movie"
    onClick={[Function]}
  >
    Skip Movie Stuff
  </button>
</article>
`;
